src/init-ceph.in

```diff
@@ -425,6 +431,52 @@ osd_has_blocked_ops() {
     fi
 }
 
+osd_has_stuck_peering() {
+    local name=$1
+    local id=$2
+
+    # Abort if we had previous errors with Ceph
+    if [ "$CEPH_FAILURE" = "true" ]; then
+        wlog $name "WARN" "Ceph cluster is marked as failed, aborting stuck peering check"
+        echo "false"; return
+    fi
+
+    # Cache Ceph Health for later use as calling Ceph takes time This is
+    # initially cached from the hang check but check and call again here if
+    # needed
+    if [ -z "$CEPH_HEALTH_DETAIL" ]; then
+        execute_ceph_cmd CEPH_HEALTH_DETAIL $name "ceph health detail"
+        if [ $? -ne 0 ]; then
+            wlog $name "WARN" "Aborting stuck peering check"
+            echo "false"; return
+        fi
+    fi
+
+    # Ignore health check if OSDs are administratively up
+    # Note this can be done with: 'ceph osd set nodown; /etc/init.d/ceph stop <osd.id>'
+    $(echo "$CEPH_HEALTH_DETAIL" | grep -q "nodown.*set")
+    if [ $? -eq 0 ]; then
+        wlog $name "WARN" "Ceph 'nodown' flag is set, aborting stuck peering check"
+        echo "false"; return
+    fi
```

Comment:
When would we want to keep the OSD adminstratively up? I get it in the case of the blocked_ops where we want to mark an OSD as administratively down and not have it restarted, but I'm not sure I understand why we want this check here? Is this for completeness, debug, or a specific scenario?

Reply:
In case of debugging you don't want the OSD to be restarted.

Reply:
OK. Tx.